configuration.go

```diff
@@ -176,7 +176,12 @@ func (v *VirtualMachineConfiguration) SetStorageDevicesVirtualMachineConfigurati
 }
 
 // SetDirectorySharingDevicesVirtualMachineConfiguration sets list of directory sharing devices. Empty by default.
+//
+// This is only supported on macOS 12 and newer. Older versions do nothing.
 func (v *VirtualMachineConfiguration) SetDirectorySharingDevicesVirtualMachineConfiguration(cs []DirectorySharingDeviceConfiguration) {
+	if macosMajorVersionLessThan(12) {
+		return
+	}
```

Comment:
In what scenario is this needed? I remember not adding version checks to these methods on the reasoning that you could not instantiate `DirectorySharingDeviceConfiguration`, so you would not really be able to call this method.

Reply:
Basically, if the object and method versions are equal, they are not checked.

In this case, the configuration object is v11 and the method is v12, so we need to check.

Throws objc when using an empty array if there is no check.